feature_layers/add-feature-clustering/src/main/java/com/esri/samples/add_feature_clustering/AddFeatureClusteringSample.java

```diff
@@ -211,83 +273,71 @@ private VBox controlsVBox() {
     });
-    displayHBox.getChildren().addAll(label1, mDisplayLabelsCheckbox);
+    displayHBox.getChildren().addAll(label1, displayLabelsCheckbox);
 
     HBox clusterRadiusHBox = new HBox(SPACING);
-    mClusterRadiusLabel = new Label("Cluster radius: " + DEFAULT_CLUSTER_R);
-    mClusterRadiusLabel.setMinWidth(MIN_WIDTH);
-    mClusterRadiusLabel.setTextFill(Color.WHITE);
-    mClusterRadiusSlider = new Slider();
-    mClusterRadiusSlider.setValue(DEFAULT_CLUSTER_R);
-
-    // create a slider for adjusting cluster radius
-    mClusterRadiusSlider = new Slider(30, 85, DEFAULT_CLUSTER_R);
-    mClusterRadiusSlider.setMaxWidth(MAX_WIDTH);
-    // add a listener to the slider's value property to set the cluster radius
-    mClusterRadiusSlider.setOnMouseReleased(event -> {
-      mClusterRadiusLabel.setText(String.valueOf("Cluster radius: " + (int) mClusterRadiusSlider.getValue()));
-      mClusteringFeatureReduction.setRadius(mClusterRadiusSlider.getValue());
+    clusterRadiusLabel = new Label("Cluster radius: ");
+    clusterRadiusLabel.setMinWidth(MIN_WIDTH);
+    clusterRadiusLabel.setTextFill(Color.WHITE);
+    clusterRadiusPicker = new ComboBox();
+    clusterRadiusPicker.getItems().addAll(30, 45, 60, 75, 90);
+    clusterRadiusPicker.setValue(DEFAULT_CLUSTER_RADIUS);
+
+    // add a listener to the ComboBox's value property to set the cluster radius
+    clusterRadiusPicker.setOnAction(event -> {
+      clusteringFeatureReduction.setRadius((int) clusterRadiusPicker.getValue());
     });
-    clusterRadiusHBox.getChildren().addAll(mClusterRadiusLabel, mClusterRadiusSlider);
+    clusterRadiusHBox.getChildren().addAll(clusterRadiusLabel, clusterRadiusPicker);
 
-    final int DEFAULT_SCALE_WIDTH = 0;
-    final int MAX_SCALE_WIDTH = 150000;
     HBox maxScaleHBox = new HBox(SPACING);
-    mMaxScaleLabel = new Label("MaxScale: " + DEFAULT_SCALE_WIDTH);
-    mMaxScaleLabel.setMinWidth(MIN_WIDTH);
-    mMaxScaleLabel.setTextFill(Color.WHITE);
-    // create a slider for adjusting max scale
-    mMaxScaleSlider = new Slider(0, MAX_SCALE_WIDTH, DEFAULT_SCALE_WIDTH);
-    mMaxScaleSlider.setMaxWidth(MAX_SCALE_WIDTH);
-    // add a listener to the slider's value property to set the max scale
-    mMaxScaleSlider.setOnMouseReleased(event -> {
-      mMaxScaleLabel.setText(String.valueOf("MaxScale: " + (int) mMaxScaleSlider.getValue()));
-      mClusteringFeatureReduction.setMaxScale(mMaxScaleSlider.getValue());
+    maxScaleLabel = new Label("MaxScale: ");
+    maxScaleLabel.setMinWidth(MIN_WIDTH);
+    maxScaleLabel.setTextFill(Color.WHITE);
+    // create a ComboBox for adjusting max scale
+    maxScalePicker = new ComboBox();
+    maxScalePicker.getItems().addAll(0, 1000, 5000, 10000, 50000, 100000, 500000);
+    maxScalePicker.setValue(0);
+
+    // add a listener to the ComboBox's value property to set the max scale
+    maxScalePicker.setOnAction(event -> {
+      clusteringFeatureReduction.setMaxScale((int) maxScalePicker.getValue());
```

Comment:
If you parameterise the declaration of the ComboBox you can remove the cast to `int`.